tests/admin/test_forms.py

```diff
@@ -123,29 +129,113 @@ def test_clean_existing_pending_link(self, existing_email):
+        super(TestEnterpriseCustomerAdminForm, self).setUp()
         self.idp_choices_mock = mock.Mock()
         patcher = mock.patch("enterprise.admin.forms.utils.get_idp_choices", self.idp_choices_mock)
         patcher.start()
         self.addCleanup(patcher.stop)
+        self.catalog_id = random.uniform(1, 1000000)
+        self.patch_enterprise_customer_user()
+        self.addCleanup(self.unpatch_enterprise_customer_user)
+
+    @staticmethod
+    def patch_enterprise_customer_user():
+        """
+        Set the class-level form user to None.
+        """
+        EnterpriseCustomerAdminForm.user = None
+
+    @staticmethod
+    def unpatch_enterprise_customer_user():
+        """
+        Set the form to its original state.
+        """
+        delattr(EnterpriseCustomerAdminForm, 'user')  # pylint: disable=literal-used-as-attribute
 
-    def test_no_idp_choices(self):
+    @mock.patch('enterprise.admin.forms.get_all_catalogs')
+    def test_no_idp_choices(self, mocked_method):
+        mocked_method.return_value = []
         self.idp_choices_mock.return_value = None
-        form = EnterpriseCustomerForm()
+        form = EnterpriseCustomerAdminForm()
         assert not hasattr(form.fields['identity_provider'], 'choices')
 
-    def test_with_idp_choices(self):
+    @mock.patch('enterprise.admin.forms.get_all_catalogs')
+    def test_with_idp_choices(self, mocked_method):
+        mocked_method.return_value = []
         choices = (('idp1', 'idp1'), ('idp2', 'idp2'))
         self.idp_choices_mock.return_value = choices
-        form = EnterpriseCustomerForm()
+        form = EnterpriseCustomerAdminForm()
         assert hasattr(form.fields['identity_provider'], 'choices')
         assert form.fields['identity_provider'].choices == list(choices)
+
+    @mock.patch('enterprise.admin.forms.get_all_catalogs')
+    def test_interface_displays_selected_option(self, mock_method):
+        mock_method.return_value = [
+            {
+                "id": self.catalog_id,
+                "name": "My Catalog"
+            }
+        ]
+        form = EnterpriseCustomerAdminForm()
+        self.assertTrue(isinstance(form.fields['catalog'], forms.ChoiceField))
+        self.assertEqual(
+            form.fields['catalog']._get_choices(),  # pylint: disable=protected-access
+            forms.ChoiceField(  # pylint: disable=protected-access
+                choices=((None, 'None'), (self.catalog_id, 'My Catalog'),)
+            )._get_choices()
+        )
+
+    @mock.patch('enterprise.course_catalog_api.get_catalog_api_client')
+    @mock.patch('enterprise.course_catalog_api.CatalogIntegration')
+    @mock.patch('enterprise.course_catalog_api.get_edx_api_data')
+    def test_with_mocked_get_edx_data(self, mocked_method, *args):  # pylint: disable=unused-argument
+        mocked_method.return_value = [
+            {
+                "id": self.catalog_id,
+                "name": "My Catalog"
+            }
```

Comment:
@haikuginger nit: would be nice to test how will it behave if more than one item is returned